components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java

```diff
@@ -884,6 +884,24 @@ private void callDefaultProvisioningHandler(String username, AuthenticationConte
         // Remove role claim from local claims as roles are specifically handled.
         localClaimValues.remove(FrameworkUtils.getLocalClaimUriMappedForIdPRoleClaim(externalIdPConfig));
         localClaimValues.remove(UserCoreConstants.USER_STORE_GROUPS_CLAIM);
+
+        Map<String, String> runtimeClaims = context.getRuntimeClaims();
+        Map<String, String> remoteClaims =
+                (Map<String, String>) context.getProperty(FrameworkConstants.UNFILTERED_IDP_CLAIM_VALUES);
+
+        // Remove or revert runtime claims from local claims before calling JIT provisioning.
+        for (Map.Entry<String, String> entry : runtimeClaims.entrySet()) {
+            String localClaimURI = entry.getKey();
+            if (claimMapping != null && remoteClaims.containsKey(claimMapping.get(localClaimURI))) {
```

Comment:
As discussed, hope you have tested 'attributeSyncMethod' set to 'NONE' and with 'Assert identity using mapped local subject identifier ', this logic is OK.

Reply:
Yes.

I tested with `attributeSyncMethod` set to `NONE` and with `Assert identity using mapped local subject identifier `.
1. When a temp claim overrides, the temp claim is shown within the session.
2. When a temp claim doesn't override, the local value is shown (not the IDP value).

This is the expected behavior.